jupyterhub/app.py

```diff
@@ -3150,7 +3150,8 @@ async def start(self):
             self.last_activity_callback = pc
             pc.start()
 
-        self.log.info("JupyterHub is now running at %s", self.proxy.public_url)
+        if self.proxy.should_start:
+            self.log.info("JupyterHub is now running at %s", self.proxy.public_url)
```

Comment:
I agree with @consideRatio that it's nice to still have the "JupyterHub is now running" marker to indicate that startup is done. How about:

```suggestion
            self.log.info("JupyterHub is now running at %s", self.proxy.public_url)
        else:
            self.log.info("JupyterHub is now running, internal Hub API at %s", self.hub.url)
```

or split "JupyterHub is now running" into its own log statement, and log the URL separately under the condition.

Reply:
Your solution looks better